tools/cleanup_caches.py

```
@@ -0,0 +1,322 @@
+        # 并发保护：最近 access_window 秒内被访问过 -> 跳过
+        if age_last_access(c) < access_window_s:
+            c["_reason"] = f"skip: last access within {access_window_s}s (possible in-flight restore)"
+            protected.append(c)
+            continue
+        final_to_delete.append(c)
+
+    # ---- 规则 4：总量水位保护 ----
+    remaining_size = sum(c.get("size_in_bytes", 0) for c in remaining)
+    deleted_size = sum(c.get("size_in_bytes", 0) for c in final_to_delete)
+    if watermark_gb and remaining_size - deleted_size > watermark_gb * 1024**3:
+        # 从旧到新删，直到低于 target
```

Ruff (0.16.2)

[warning] Comment contains ambiguous `，` (FULLWIDTH COMMA). Did you mean `,` (COMMA)?

(RUF003)